olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -569,30 +567,30 @@ client::CancellationToken AuthenticationClient::Impl::SignUpHereUser(
   std::shared_ptr<std::stringstream> payload =
       std::make_shared<std::stringstream>();
   request.WithBody(generateSignUpBody(properties));
-  auto send_outcome = network_->Send(
-      request, payload,
-      [callback, payload,
-       credentials](const http::NetworkResponse& network_response) {
-        auto response_status = network_response.GetStatus();
-        auto error_msg = network_response.GetError();
-
-        if (response_status < 0) {
-          // Network error response
-          AuthenticationError error(response_status, error_msg);
-          callback(error);
-          return;
-        }
-
-        std::shared_ptr<Document> document = std::make_shared<Document>();
-        IStreamWrapper stream(*payload.get());
-        document->ParseStream(stream);
-
-        std::shared_ptr<SignUpResultImpl> resp_impl =
-            std::make_shared<SignUpResultImpl>(response_status, error_msg,
-                                               document);
-        SignUpResult response(resp_impl);
-        callback(response);
-      });
+  auto send_outcome =
+      network_->Send(request, payload,
+                     [callback, payload, credentials](
+                         const http::NetworkResponse& network_response) {
+                       auto response_status = network_response.GetStatus();
+                       auto error_msg = network_response.GetError();
+
+                       if (response_status < 0) {
+                         // Network error response
+                         AuthenticationError error(response_status, error_msg);
```

Comment:
```suggestion
                         callback({{response_status, error_msg}});
```
Similar is used in CatalogClient.